ui/src/pages/gcp/login.js

```diff
@@ -303,62 +298,61 @@ class GCPLoginPage extends React.Component {
-                            <img src='/static/google-cloud.svg'></img>
-                        </GcpHeader>
-                        <ParentWrapperDiv>
-                            <h3>
-                                Error: There are no GCP project roles associated
-                                with your account.
-                            </h3>
-                            <p>
-                                Check to make sure that your <tt>gcp.*</tt>{' '}
-                                roles contain your user.
-                                <br />
-                                E.g., if your account is 'jdoe', then{' '}
-                                <tt>gcp.fed.admin.user</tt> should have{' '}
-                                <tt>{USER_DOMAIN}.jdoe</tt> as a member.
-                            </p>
-                            <p>
-                                If that does not work, please ask your Athenz
-                                domain admin for assistance.
-                            </p>
-                        </ParentWrapperDiv>
-                    </div>
-                </CacheProvider>
-            );
-        }
-
-        return (
+        let gcpLoginContainer = displayProjects.length ? (
+            <div data-testid='gcp-login'>
+                <GcpHeader>
+                    <img src='/static/google-cloud.svg'></img>
+                </GcpHeader>
+                <ParentWrapperDiv>
+                    <form action='/gcp/login/post' method='post'>
+                        <input
+                            type='hidden'
+                            name='_csrf'
+                            value={this.props._csrf}
+                        ></input>
+                        <input
+                            type='hidden'
+                            name='isAdmin'
+                            value={this.props.isAdmin}
+                        ></input>
+                        <StyledP>Select a role:</StyledP>
+                        {displayProjects}
+                        <SubmitContainer>
+                            <Button type={'submit'}>Submit</Button>
+                        </SubmitContainer>
+                    </form>
+                </ParentWrapperDiv>
+            </div>
+        ) : (
+            <div data-testid='gcp-login-error'>
+                <GcpHeader>
+                    <img src='/static/google-cloud.svg'></img>
+                </GcpHeader>
+                <ParentWrapperDiv>
+                    <h3>
+                        Error: There are no GCP project roles associated with
+                        your account.
+                    </h3>
+                    <p>
+                        Check to make sure that your <tt>gcp.*</tt> roles
+                        contain your user.
+                        <br />
+                        E.g., if your account is 'jdoe', then{' '}
+                        <tt>gcp.fed.admin.user</tt> should have{' '}
```

Comment:
this is a yahoo specific role so we should not included it here. Let's keep it generic and just refer to that your account should be in the configured federated roles that have access to the project and not list any specific role names.

Reply:
how's this? 
<img width="1722" alt="image" src="https://github.com/AthenZ/athenz/assets/25251812/0f3abdb4-2f82-4f8f-8973-955831f853b9">
```
Error: There are no GCP project roles associated with your account.
Check to make sure that your gcp.* roles contain your user. Your account should be in the configured federated roles that have access to the project. If that does not work, please ask your Athenz domain admin for assistance.
```